sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveCatalogs.scala

```diff
@@ -167,35 +169,36 @@ class ResolveCatalogs(val catalogManager: CatalogManager)
         writeOptions = c.options.filterKeys(_ != "path"),
         orCreate = c.orCreate)
 
-    case DropTableStatement(NonSessionCatalog(catalog, tableName), ifExists, _) =>
+    case DropTableStatement(NonSessionCatalogAndTable(catalog, tableName), ifExists, _) =>
       DropTable(catalog.asTableCatalog, tableName.asIdentifier, ifExists)
 
-    case DropViewStatement(NonSessionCatalog(catalog, viewName), _) =>
+    case DropViewStatement(NonSessionCatalogAndTable(catalog, viewName), _) =>
       throw new AnalysisException(
         s"Can not specify catalog `${catalog.name}` for view ${viewName.quoted} " +
           s"because view support in catalog has not been implemented yet")
 
-    case c @ CreateNamespaceStatement(NonSessionCatalog(catalog, nameParts), _, _) =>
+    case c @ CreateNamespaceStatement(NonSessionCatalogAndNamespace(catalog, namespace), _, _) =>
       CreateNamespace(
         catalog.asNamespaceCatalog,
-        nameParts,
+        namespace,
         c.ifNotExists,
         c.properties)
 
-    case DropNamespaceStatement(NonSessionCatalog(catalog, nameParts), ifExists, cascade) =>
-      DropNamespace(catalog, nameParts, ifExists, cascade)
+    case DropNamespaceStatement(NonSessionCatalogAndNamespace(catalog, ns), ifExists, cascade) =>
```

Comment:
nit: let's be consistent and always use `ns`

Reply:
and also `tbl` instead of `tableName`, to make the code shorter.

Reply:
Updated. I also applied this to `ResolveSessionCatalogs.scala`. Please let me know if that was not intended, and I can revert.